plans/elasticsearch/config/elasticsearch.yml

```diff
@@ -1,16 +1,70 @@
 
-cluster.name: {{cfg.cluster-name}}
-node.name: {{cfg.node-name}}
-path.data: {{cfg.path-data}}
-path.logs: {{cfg.path-logs}}
-path.plugins: {{cfg.path-plugins}}
-path.scripts: {{cfg.path-scripts}}
-bootstrap.mlockall: {{cfg.bootstrap-mlockall}}
-network.host: {{cfg.network-host}}
-http.port: {{cfg.network-port}}
-discovery.zen.ping.multicast.enabled: {{cfg.discovery-ping_multicast_enabled}}
-discovery.zen.ping.unicast.hosts: {{cfg.discovery-ping_unicast_hosts}}
-discovery.zen.minimum_master_nodes: {{cfg.discovery-minimum_master_nodes}}
-gateway.recover_after_nodes: {{cfg.gateway-recover_after_nodes}}
-node.max_local_storage_nodes: {{cfg.node-max_local_storage_nodes}}
-action.destructive_requires_name: {{cfg.action-destructive_requires_name}}
+cluster.name: {{cfg.cluster.name}}
+
+cluster.routing.allocation.node_concurrent_recoveries: {{cfg.cluster.routing.allocation.node_concurrent_recoveries}}
+cluster.routing.allocation.node_initial_primaries_recoveries: {{cfg.cluster.routing.allocation.node_initial_primaries_recoveries}}
+cluster.routing.allocation.same_shard.host: {{cfg.cluster.routing.allocation.same_shard-host}}
+
+{{~#if cfg.cluster.routing.allocation.awareness-attributes}}
+cluster.routing.allocation.awareness.attributes: {{cfg.cluster.routing.allocation.awareness-attributes}}
+{{~/if}}
+
+node.name: {{cfg.node.name}}
+node.max_local_storage_nodes: {{cfg.node.max_local_storage_nodes}}
+
+{{~#if cfg.node.rack_id}}
+node.rack_id: {{cfg.node.rack_id}}
+{{~/if}}
+{{~#if cfg.node.zone}}
+node.zone: {{cfg.node.zone}}
+{{~/if}}
+{{~#if cfg.node.master}}
+node.master: {{cfg.node.master}}
+{{~/if}}
+{{~#if cfg.node.data}}
+node.data: {{cfg.node.data}}
+{{~/if}}
+
+
+path.data: {{cfg.path.data}}
+path.logs: {{cfg.path.logs}}
+path.plugins: {{cfg.path.plugins}}
+path.scripts: {{cfg.path.scripts}}
+
+indices.recovery.concurrent_streams: {{cfg.indices.recovery.concurrent_streams}}
+indices.recovery.concurrent_small_file_streams: {{cfg.indices.recovery.concurrent_small_file_streams}}
+indices.recovery.file_chunk_size: {{cfg.indices.recovery.file_chunk_size}}
+indices.recovery.translog_ops: {{cfg.indices.recovery.translog_ops}}
+indices.recovery.translog_size: {{cfg.indices.recovery.translog_size}}
+indices.recovery.compress: {{cfg.indices.recovery.compress}}
+indices.recovery.max_bytes_per_sec: {{cfg.indices.recovery.max_bytes_per_sec}}
+
+{{~#if cfg.indices.fielddata.cache-size}}
+indices.fielddata.cache.size: {{cfg.indices.fielddata.cache-size}}
+{{~/if}}
+
+indices.breaker.total.limit: {{cfg.indices.breaker.total-limit}}
```

Comment:
I think this is a good improvement. I'm :+1:

On Fri, Jun 17, 2016, 1:13 PM xcxsxvx <notifications@github.com> wrote:

> In plans/elasticsearch/config/elasticsearch.yml
> <https://github.com/habitat-sh/habitat/pull/896#discussion_r67552923>:
>
> > +path.plugins: {{cfg.path.plugins}}
> > +path.scripts: {{cfg.path.scripts}}
> > +
> > +indices.recovery.concurrent_streams: {{cfg.indices.recovery.concurrent_streams}}
> > +indices.recovery.concurrent_small_file_streams: {{cfg.indices.recovery.concurrent_small_file_streams}}
> > +indices.recovery.file_chunk_size: {{cfg.indices.recovery.file_chunk_size}}
> > +indices.recovery.translog_ops: {{cfg.indices.recovery.translog_ops}}
> > +indices.recovery.translog_size: {{cfg.indices.recovery.translog_size}}
> > +indices.recovery.compress: {{cfg.indices.recovery.compress}}
> > +indices.recovery.max_bytes_per_sec: {{cfg.indices.recovery.max_bytes_per_sec}}
> > +
> > +{{~#if cfg.indices.fielddata.cache-size}}
> > +indices.fielddata.cache.size: {{cfg.indices.fielddata.cache-size}}
> > +{{~/if}}
> > +
> > +indices.breaker.total.limit: {{cfg.indices.breaker.total-limit}}
>
> Well, anything Habitat doesn't expose as a "cfg." variable won't be
> changeable by the end user, right?
> I figured you'd want to expose as much configuration as possible and let
> the user choose which, if any, to change.
> Some of these are changeable dynamically through the ES API, but I'd
> rather have them set the way I want them at startup.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/habitat-sh/habitat/pull/896/files/3a0037c27d3416e263f0d4b9592f3c3d0b92d01b#r67552923>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABE_bTqLV0a5QkklJ9MalkWbCBJh_f1hks5qMuPKgaJpZM4I33iN>
> .
>